### PR TITLE
Fix CMakeLists typos for Eigen

### PR DESCRIPTION
**Pull request by kutukvpavel**
# Checklist

- [x] Component contains License
- [x] Component contains README.md
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [x] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [ ] _Optional:_ Component contains unit tests
- [x] CI passing

# Change description
Typos in lines 17 and 19 of root CMakeLists.txt for Eigen (optimization definitions missing CONFIG_ name prefix) cause build failure (with an unhelpful error message "Unsupported optimization level"). The issue only manifests if optimization levels other than DEFAULT or SIZE are used (for example, DEBUG), which is likely the reason why it was overlooked initially.

This pull request fixes the typos, replacing mistyped optimization definitions with correct ones.

### CODE REVIEW

**Review by igrr — APPROVED** [review]

File: eigen/CMakeLists.txt [hunk 1]
```diff
@@ -14,9 +14,9 @@ if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
     set(opt_args -DCMAKE_BUILD_TYPE=Debug)
 elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
     set(opt_args -DCMAKE_BUILD_TYPE=Relesee)
-elseif(COMPILER_OPTIMIZATION_PERF)
+elseif(CONFIG_COMPILER_OPTIMIZATION_PERF)
     set(opt_args -DCMAKE_BUILD_TYPE=Relesee)
```
Comment (igrr) [inline eigen/CMakeLists.txt:18]: @kutukvpavel Could you please also fix another typo here (should be `Release`). And in the line above I believe it should be `MinSizeRel`

Comment (kutukvpavel) [reply eigen/CMakeLists.txt]: @igrr No problem. Fixed.

**Comment by igrr** [comment]
@kutukvpavel Thanks! I forgot one more thing (sorry!) — please bump the version in idf_component.yml so that the new release to the component registry happens.
https://github.com/espressif/idf-extra-components/blob/12a599fcc73729d6116fc49d535290aab347646e/eigen/idf_component.yml#L2
need to change this to `version: 3.4.0~1`.

Also please combine your changes into one commit, if possible (`git rebase -i`).

**Comment by kutukvpavel** [comment]
@igrr Done.

**Pull request merged**